ethportal-api/src/types/execution/header.rs

```diff
@@ -157,16 +183,31 @@ impl Decodable for Header {
             nonce: Some(rlp.val_at(14)?),
             base_fee_per_gas: None,
             withdrawals_root: None,
+            blob_gas_used: None,
+            excess_blob_gas: None,
+            parent_beacon_block_root: None,
         };
 
-        if header.number >= LONDON_BLOCK_NUMBER {
+        if rlp.at(15).is_ok() {
```

Comment:
```rust
if let Ok(val) = rlp.at(15) {
    header.base_fee_per_gas = Some(val)
}
```

same goes for the following statements

Reply:
I guess this does assume that the `Header` json values are correctly constructed. Aka, no pre-london headers will contain `base_fee_per_gas`, which I think is a safe assumption. And if a source header is incorrectly constructed, we'll know when we try to hash it